pandas/core/internals/blocks.py

```diff
@@ -2002,6 +2002,18 @@ class ObjectValuesExtensionBlock(ExtensionBlock):
     def external_values(self):
         return self.values.astype(object)
 
+    def _can_hold_element(self, element: Any) -> bool:
+        if is_valid_nat_for_dtype(element, self.dtype):
+            return True
+        if element is NaT:
```

Comment:
@jbrockmendel ok here?

@samc1213 are all of these branches actually hit? in particular why L2008 and L2010

Reply:
@jreback Added these due to unit tests. Specifically, L2010 is required for `pandas/tests/extension/test_interval.py::TestSetitem::test_setitem_empty_indxer` to pass. L2008 may not be required, I just removed it and will see if the tests pass...